src/widgets/FormControl.js

```diff
@@ -223,23 +250,29 @@ const FormControlComponent = ({
       <Buttons />
     </View>
   );
+
+  return isConfirmFormOpen ? InputConfirm : InputDetails;
 };
 
 const mapDispatchToProps = (dispatch, ownProps) => {
   const initialiseForm = config => dispatch(FormActions.initialiseForm(config));
+  const showConfirmForm = () => dispatch(FormActions.showConfirmForm());
   const onUpdateForm = (key, value) => dispatch(FormActions.updateForm(key, value));
-  const onCancel = () => {
-    dispatch(FormActions.resetForm());
-    ownProps.onCancel();
-  };
-  return { initialiseForm, onUpdateForm, onCancel };
+  const onCancel = () => ownProps.onCancel() && dispatch(FormActions.resetForm());
+  return { initialiseForm, showConfirmForm, onUpdateForm, onCancel };
 };
 
-const mapStateToProps = state => {
+const mapStateToProps = (state, ownProps) => {
   const form = selectForm(state);
   const canSaveForm = selectCanSaveForm(state);
+  const isConfirmFormOpen = selectIsConfirmFormOpen(state);
   const completedForm = selectCompletedForm(state);
-  return { form, canSaveForm, completedForm };
+  const { confirmOnSave } = ownProps;
+  console.log('FORM:', form);
+  console.log('CANSAVEFORM:', canSaveForm);
+  console.log('ISCONFIRMFORMOPEN:', isConfirmFormOpen);
+  console.log('COMPLETEDFORM:', completedForm);
```

Comment:
![chef](https://user-images.githubusercontent.com/35858975/79163040-61a6b000-7e32-11ea-81a9-b58b0b1e5f18.gif)

Reply:
Oops, forgot to remove my awesome new SNIPPET.

Reply:
6533a87bbf31d6eefcdcfeac7ab25515a99b4ac7.